src/HttpBodyStream.php

```diff
@@ -64,6 +64,11 @@ public function close()
 
         $this->closed = true;
 
+        $this->input->removeListener('data', array($this, 'handleData'));
+        $this->input->removeListener('end', array($this, 'handleEnd'));
+        $this->input->removeListener('error', array($this, 'handleError'));
+        $this->input->removeListener('close', array($this, 'close'));
```

Comment:
Not sure who's responsible here. The input stream will be closed 2 lines down. Shouldn't the input stop emitting events after it is closed?

If not, isn't the input misbehaving here? (and this should be fixed instead?)